### PR TITLE
Necessary changes that presented themselves while fixing up the example apps

### PR DESCRIPTION
**Pull request by briandilley**
## Description

Necessary changes that presented themselves while fixing up the example apps.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-jvm-sdk/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by briandilley** [comment]
Sorry for using the same branch as last time - it muddies up the other PRs.

File: flow.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (psiemens) [inline flow.json:1]: Just want to check - did you mean to commit this file?

File: .github/workflows/ci-publish-snapshot.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Build
+        run: ./gradlew --warning-mode all build -x test
+
+      - name: Test
+        run: ./gradlew --no-daemon --max-workers=2 --warning-mode=all test --continue
+        if: "!contains(github.event.head_commit.message, '[i am bad and i hate unit tests]')"
+
+      - name: Publish snapshot
+        run: ./gradlew -x test -PsonatypeUsername=${{ FLOW_JVM_SDK_SONATYPE_USERNAME }} -PsonatypePassword=${{ FLOW_JVM_SDK_SONATYPE_PASSWORD }} -PgroupId=${{ FLOW_JVM_SDK_SONATYPE_GROUP_ID }} publishToSonatype closeAndReleaseSonatypeStagingRepository
```
Comment (psiemens) [inline .github/workflows/ci-publish-snapshot.yml:33]: I assume we (Flow team) need to fill in these secrets?

**Review by psiemens — APPROVED** [review]

File: flow.json (hunk 1 above)
Comment (briandilley) [reply flow.json:1]: yes, it's for the unit tests

File: .github/workflows/ci-publish-snapshot.yml (hunk 2 above)
Comment (briandilley) [reply .github/workflows/ci-publish-snapshot.yml]: Yes - i updated it to use the github secrets (that was my intention from the beginning)

File: build.gradle.kts [hunk 3]
```diff
@@ -26,72 +56,105 @@ dependencies {
...
-    mavenCentral()
-    maven { url = uri("https://jitpack.io") }
-}
+    test {
+        useJUnitPlatform()
+        testLogging {
+            exceptionFormat 	= org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
+            showExceptions 		= true
+            showStackTraces 	= true
+            showCauses 			= true
```
Comment (psiemens) [inline build.gradle.kts:67]: Spacing's a bit off here

File: .github/workflows/ci-publish-release.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      - id: get_version
+        uses: battila7/get-version-action@v2
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: '11'
```
Comment (psiemens) [inline .github/workflows/ci-publish-release.yml:22]: I noticed that the gradle build target is `1.8` -- should this be `8` as well, or does it not matter?

Comment (briandilley) [reply .github/workflows/ci-publish-release.yml:22]: It doesn't matter - the compilers are all backward compatible (ie: you can target older bytecode versions)

File: build.gradle.kts (hunk 3 above)
Comment (briandilley) [reply build.gradle.kts:67]: Will address that

**Review by psiemens — APPROVED** [review]

**Comment by briandilley** [comment]
I'm going to close/cancel this pull request and create a new one in a bit that contains some CI/CD stuff.

**Pull request closed without merging**